src/test/java/com/endava/internship/collections/StudentListTest.java

```diff
@@ -0,0 +1,870 @@
+package com.endava.internship.collections;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class StudentListTest {
+
+    @BeforeEach
+    public void setUp() {
+    }
+
+    @Test
+    void constructorShouldThrowAnExceptionIfTheCapacityIsLessThenZero() {
+        assertThrows(IllegalArgumentException.class, () -> new StudentList<Student>(-1));
+    }
+
+    @Test
+    void shouldTheSizeBeRight() {
+        //given
+        List<Student> studentList = new StudentList<>();
+        Student st1 = new Student("Unu", LocalDate.of(2001, 1, 1), "Student ONE");
+        Student st2 = new Student("Doi", LocalDate.of(2004, 1, 1), "Student TWO");
+        Student st3 = new Student("Trei", LocalDate.of(2006, 1, 1), "Student TREE");
+        studentList.add(st1);
+        studentList.add(st2);
+        studentList.add(st3);
```

Comment:
duplicate code.
many of your tests use the same or similar data.
extract the studentList  at instance level
make use of @BeforeEach @AfterEach, private utility methods to populate test data

Reply:
Done.